### ISSUE TITLE
Stumpwm crashes when turning off an output with xrandr

### ISSUE THREAD

**Issue by scottjad** [stumpwm/stumpwm#586]
# Summary
When I transition from dual monitors to a single monitor, stumpwm crashes.

I believe this is the result of the new randr commits in stumpwm (@ralt), as it started when I updated stumpwm from a version that was a couple months old. Also, if I comment out the randr branch of the cond in make-screen-heads in heads.lisp it will not crash.

I have temporarily fixed the problem for myself by changing this in output->head:

```
(and (eq request-status :success)
     (eq status :connected))
```
to:

```
(and (eq request-status :success)
     (eq status :connected)
     (not (= crtc 0)))
```

but I don't understand the meaning of crtc equalling 0 so I'm not sure if that's a correct fix.

When I turn a monitor off with xrandr, xlib:rr-get-output-info will still return :success and :connected, but the crtc will be 0, and calling xlib:rr-get-crtc-info with that 0 raises an error, I believe.

# Shell command that causes the crash

This script disables all but my main monitor.

```
MAIN=DP-0
SECONDARY=DVI-D-0
TERTIARY=HDMI-0
xrandr --output $MAIN --primary --auto --output $SECONDARY --off --output $TERTIARY --off
```

# Backtrace
```
  Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {1005C286D3}>
0: (SB-VM::TAIL-CALL-SYMBOL #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)> #S(XLIB::REPLY-BUFFER :SIZE 32 :IBUF8 #(0 148 14 129 0 0 0 0 20 0 140 0 ...) :NEXT NIL :DATA-SIZE 32))
1: (XLIB::MAKE-ERROR #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)> #S(XLIB::REPLY-BUFFER :SIZE 32 :IBUF8 #(0 148 14 129 0 0 0 0 20 0 140 0 ...) :NEXT NIL :DATA-SIZE 32) NIL)
2: (XLIB::READ-REPLY #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)> #S(XLIB::PENDING-COMMAND :SEQUENCE 33038 :REPLY-BUFFER NIL :PROCESS #<SB-THREAD:THREAD "main thread" RUNNING {1005C286D3}> :NEXT NIL))
3: (XLIB:RR-GET-CRTC-INFO #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)> 0 3761577954 LIST)
4: (STUMPWM::OUTPUT->HEAD 446 0)
5: (STUMPWM::MAKE-SCREEN-RANDR-HEADS #<XLIB:WINDOW :0 1E4>)
6: ((LAMBDA (&REST #:G0 &KEY :STACK-MODE :WINDOW :X :Y :WIDTH :HEIGHT :BORDER-WIDTH :VALUE-MASK &ALLOW-OTHER-KEYS) :IN "/home/scott/src/stumpwm/events.lisp") :STACK-MODE NIL :WINDOW #<XLIB:WINDOW :0 1E4> :X 0 :Y 0 :WIDTH 3640 :HEIGHT 1920 :BORDER-WIDTH 0 :VALUE-MASK NIL)
7: (STUMPWM::HANDLE-EVENT :DISPLAY #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)> :EVENT-KEY :CONFIGURE-NOTIFY :EVENT-CODE 22 :SEND-EVENT-P NIL :SEQUENCE 33033 :EVENT-WINDOW #<XLIB:WINDOW :0 1E4> :WINDOW #<XLIB:WINDOW :0 1E4> :ABOVE-SIBLING NIL :X 0 :Y 0 :WIDTH 3640 :HEIGHT 1920 :BORDER-WIDTH 0 :OVERRIDE-REDIRECT-P NIL)
8: ((FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN XLIB:PROCESS-EVENT))
9: ((FLET "WITHOUT-INTERRUPTS-BODY-29" :IN SB-THREAD::CALL-WITH-RECURSIVE-LOCK))
10: (SB-THREAD::CALL-WITH-RECURSIVE-LOCK #<CLOSURE (FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN XLIB:PROCESS-EVENT) {7FFFF63E703B}> #<SB-THREAD:MUTEX "CLX Event Lock" owner: #<SB-THREAD:THREAD "main thread" RUNNING {1005C286D3}>> T NIL)
11: ((FLET SB-IMPL::TIMEOUT-BODY :IN XLIB:PROCESS-EVENT))
12: (XLIB:PROCESS-EVENT #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)> :HANDLER #<FUNCTION STUMPWM::HANDLE-EVENT> :TIMEOUT 0 :PEEK-P NIL :DISCARD-P NIL :FORCE-OUTPUT-P T)
13: ((FLET "WITHOUT-INTERRUPTS-BODY-29" :IN SB-THREAD::CALL-WITH-RECURSIVE-LOCK))
14: (SB-THREAD::CALL-WITH-RECURSIVE-LOCK #<CLOSURE (FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN STUMPWM::DISPATCH-ALL) {7FFFF63E74AB}> #<SB-THREAD:MUTEX "CLX Event Lock" owner: #<SB-THREAD:THREAD "main thread" RUNNING {1005C286D3}>> T NIL)
15: ((FLET STUMPWM::DISPATCH-ALL :IN "/home/scott/src/stumpwm/stumpwm.lisp") #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)>)
16: ((:METHOD IO-LOOP (STUMPWM::SBCL-IO-LOOP)) #<STUMPWM::SBCL-IO-LOOP {1005526DF3}> :DESCRIPTION "StumpWM") [fast-method]
17: (STUMPWM::STUMPWM-INTERNAL-LOOP)
18: (STUMPWM::STUMPWM-INTERNAL ":0")
19: (STUMPWM ":0")
20: ((LAMBDA NIL :IN "/home/scott/src/stumpwm/make-image.lisp"))
21: ((FLET SB-UNIX::BODY :IN SB-EXT:SAVE-LISP-AND-DIE))
22: ((FLET "WITHOUT-INTERRUPTS-BODY-36" :IN SB-EXT:SAVE-LISP-AND-DIE))
23: ((LABELS SB-IMPL::RESTART-LISP :IN SB-EXT:SAVE-LISP-AND-DIE))
Caught 'The function COMMON-LISP:NIL is undefined.' at the top level. Please report this.
Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {1005C286D3}>
0: (SB-VM::TAIL-CALL-SYMBOL #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)> #S(XLIB::REPLY-BUFFER :SIZE 32 :IBUF8 #(0 148 14 129 0 0 0 0 20 0 140 0 ...) :NEXT NIL :DATA-SIZE 32))
1: (XLIB::MAKE-ERROR #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)> #S(XLIB::REPLY-BUFFER :SIZE 32 :IBUF8 #(0 148 14 129 0 0 0 0 20 0 140 0 ...) :NEXT NIL :DATA-SIZE 32) NIL)
2: (XLIB::READ-REPLY #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)> #S(XLIB::PENDING-COMMAND :SEQUENCE 33038 :REPLY-BUFFER NIL :PROCESS #<SB-THREAD:THREAD "main thread" RUNNING {1005C286D3}> :NEXT NIL))
3: (XLIB:RR-GET-CRTC-INFO #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)> 0 3761577954 LIST)
4: (STUMPWM::OUTPUT->HEAD 446 0)
5: (STUMPWM::MAKE-SCREEN-RANDR-HEADS #<XLIB:WINDOW :0 1E4>)
6: ((LAMBDA (&REST #:G0 &KEY :STACK-MODE :WINDOW :X :Y :WIDTH :HEIGHT :BORDER-WIDTH :VALUE-MASK &ALLOW-OTHER-KEYS) :IN "/home/scott/src/stumpwm/events.lisp") :STACK-MODE NIL :WINDOW #<XLIB:WINDOW :0 1E4> :X 0 :Y 0 :WIDTH 3640 :HEIGHT 1920 :BORDER-WIDTH 0 :VALUE-MASK NIL)
7: (STUMPWM::HANDLE-EVENT :DISPLAY #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)> :EVENT-KEY :CONFIGURE-NOTIFY :EVENT-CODE 22 :SEND-EVENT-P NIL :SEQUENCE 33033 :EVENT-WINDOW #<XLIB:WINDOW :0 1E4> :WINDOW #<XLIB:WINDOW :0 1E4> :ABOVE-SIBLING NIL :X 0 :Y 0 :WIDTH 3640 :HEIGHT 1920 :BORDER-WIDTH 0 :OVERRIDE-REDIRECT-P NIL)
8: ((FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN XLIB:PROCESS-EVENT))
9: ((FLET "WITHOUT-INTERRUPTS-BODY-29" :IN SB-THREAD::CALL-WITH-RECURSIVE-LOCK))
10: (SB-THREAD::CALL-WITH-RECURSIVE-LOCK #<CLOSURE (FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN XLIB:PROCESS-EVENT) {7FFFF63E703B}> #<SB-THREAD:MUTEX "CLX Event Lock" owner: #<SB-THREAD:THREAD "main thread" RUNNING {1005C286D3}>> T NIL)
11: ((FLET SB-IMPL::TIMEOUT-BODY :IN XLIB:PROCESS-EVENT))
12: (XLIB:PROCESS-EVENT #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)> :HANDLER #<FUNCTION STUMPWM::HANDLE-EVENT> :TIMEOUT 0 :PEEK-P NIL :DISCARD-P NIL :FORCE-OUTPUT-P T)
13: ((FLET "WITHOUT-INTERRUPTS-BODY-29" :IN SB-THREAD::CALL-WITH-RECURSIVE-LOCK))
14: (SB-THREAD::CALL-WITH-RECURSIVE-LOCK #<CLOSURE (FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN STUMPWM::DISPATCH-ALL) {7FFFF63E74AB}> #<SB-THREAD:MUTEX "CLX Event Lock" owner: #<SB-THREAD:THREAD "main thread" RUNNING {1005C286D3}>> T NIL)
15: ((FLET STUMPWM::DISPATCH-ALL :IN "/home/scott/src/stumpwm/stumpwm.lisp") #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)>)
16: ((:METHOD IO-LOOP (STUMPWM::SBCL-IO-LOOP)) #<STUMPWM::SBCL-IO-LOOP {1005526DF3}> :DESCRIPTION "StumpWM") [fast-method]
17: (STUMPWM::STUMPWM-INTERNAL-LOOP)
18: (STUMPWM::STUMPWM-INTERNAL ":0")
19: (STUMPWM ":0")
20: ((LAMBDA NIL :IN "/home/scott/src/stumpwm/make-image.lisp"))
21: ((FLET SB-UNIX::BODY :IN SB-EXT:SAVE-LISP-AND-DIE))
22: ((FLET "WITHOUT-INTERRUPTS-BODY-36" :IN SB-EXT:SAVE-LISP-AND-DIE))
23: ((LABELS SB-IMPL::RESTART-LISP :IN SB-EXT:SAVE-LISP-AND-DIE))

debugger invoked on a UNDEFINED-FUNCTION in thread
#<THREAD "main thread" RUNNING {1005C286D3}>:
  The function COMMON-LISP:NIL is undefined.

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

restarts (invokable by number or by possibly-abbreviated name):
  0: [CONTINUE      ] Retry calling NIL.
  1: [USE-VALUE     ] Call specified function.
  2: [RETURN-VALUE  ] Return specified values.
  3: [RETURN-NOTHING] Return zero values.
  4: [QUIT-STUMPWM  ] Quit Stumpwm
  5: [ABORT         ] Exit from the current thread.

(SB-VM::TAIL-CALL-SYMBOL #<XLIB:DISPLAY :0 (The X.Org Foundation R11906000)> #S(XLIB::REPLY-BUFFER :SIZE 32 :IBUF8 #(0 148 57 0 0 0 0 0 20 0 140 0 ...) :NEXT NIL :DATA-SIZE 32))
```
# xrandr output in dual screen mode
```
> xrandr
Screen 0: minimum 8 x 8, current 3640 x 1920, maximum 32767 x 32767
DVI-D-0 connected 1080x1920+0+0 left (normal left inverted right x axis y axis) 509mm x 286mm
   1920x1080     60.00*+
   1680x1050     59.95
   1600x1200     60.00
   1440x900      59.89
   1400x1050     59.98
   1280x1024     75.02    60.02
   1280x960      60.00
   1152x864      75.00
   1024x768      75.03    70.07    60.00
   800x600       75.00    72.19    60.32    56.25
   640x480       75.00    72.81    59.94
HDMI-0 disconnected (normal left inverted right x axis y axis)
DP-0 connected primary 2560x1440+1080+0 (normal left inverted right x axis y axis) 597mm x 336mm
   2560x1440     59.95*+
   2048x1080     60.00    24.00
   1920x1080     60.00    59.94    50.00
   1600x1200     60.00
   1280x1024     75.02    60.02
   1280x720      60.00    59.94    50.00
   1152x864      75.00
   1024x768      75.03    60.00
   800x600       75.00    60.32
   720x576       50.00
   720x480       59.94
   640x480       75.00    59.94    59.93
DP-1 disconnected (normal left inverted right x axis y axis)
```
# xrandr output in single screen mode
```
> xrandr
Screen 0: minimum 8 x 8, current 2560 x 1440, maximum 32767 x 32767
DVI-D-0 connected (normal left inverted right x axis y axis)
   1920x1080     60.00 +
   1680x1050     59.95  
   1600x1200     60.00  
   1440x900      59.89  
   1400x1050     59.98  
   1280x1024     75.02    60.02  
   1280x960      60.00  
   1152x864      75.00  
   1024x768      75.03    70.07    60.00  
   800x600       75.00    72.19    60.32    56.25  
   640x480       75.00    72.81    59.94  
HDMI-0 disconnected (normal left inverted right x axis y axis)
DP-0 connected primary 2560x1440+0+0 (normal left inverted right x axis y axis) 597mm x 336mm
   2560x1440     59.95*+
   2048x1080     60.00    24.00  
   1920x1080     60.00    59.94    50.00  
   1600x1200     60.00  
   1280x1024     75.02    60.02  
   1280x720      60.00    59.94    50.00  
   1152x864      75.00  
   1024x768      75.03    60.00  
   800x600       75.00    60.32  
   720x576       50.00  
   720x480       59.94  
   640x480       75.00    59.94    59.93  
DP-1 disconnected (normal left inverted right x axis y axis)
```
# repl when in dual screen mode
```
STUMPWM> (xlib:rr-get-screen-resources (screen-root cs))
3928284
6785
(442 443 444 445)
(446 466 467 479)
(#<XLIB::RR-MODE-INFO {10052C6233}> #<XLIB::RR-MODE-INFO {10052C62D3}>
 #<XLIB::RR-MODE-INFO {10052C6373}> #<XLIB::RR-MODE-INFO {10052C6413}>
 #<XLIB::RR-MODE-INFO {10052C64B3}> #<XLIB::RR-MODE-INFO {10052C6553}>
 #<XLIB::RR-MODE-INFO {10052C65F3}> #<XLIB::RR-MODE-INFO {10052C6693}>
 #<XLIB::RR-MODE-INFO {10052C6733}> #<XLIB::RR-MODE-INFO {10052C67D3}>
 #<XLIB::RR-MODE-INFO {10052C6873}> #<XLIB::RR-MODE-INFO {10052C6913}>
 #<XLIB::RR-MODE-INFO {10052C69B3}> #<XLIB::RR-MODE-INFO {10052C6A53}>
 #<XLIB::RR-MODE-INFO {10052C6AF3}> #<XLIB::RR-MODE-INFO {10052C6B93}>
 #<XLIB::RR-MODE-INFO {10052C6C33}> #<XLIB::RR-MODE-INFO {10052C6CD3}>
 #<XLIB::RR-MODE-INFO {10052C6D73}> #<XLIB::RR-MODE-INFO {10052C6E13}>
 #<XLIB::RR-MODE-INFO {10052C6EB3}> #<XLIB::RR-MODE-INFO {10052C6F53}>
 #<XLIB::RR-MODE-INFO {10052C6FF3}> #<XLIB::RR-MODE-INFO {10052C7093}>
 #<XLIB::RR-MODE-INFO {10052C7133}> #<XLIB::RR-MODE-INFO {10052C71D3}>
 #<XLIB::RR-MODE-INFO {10052C7273}> #<XLIB::RR-MODE-INFO {10052C7313}>
 #<XLIB::RR-MODE-INFO {10052C73B3}> #<XLIB::RR-MODE-INFO {10052C7453}>)
"1920x10801680x10501600x12001440x9001400x10501280x10241280x10241280x9601152x8641024x7681024x7681024x768800x600800x600800x600800x600640x480640x480640x4802560x14402048x10802048x10801920x10801920x10801280x7201280x7201280x720720x576720x480640x480"
STUMPWM> (output->head 446 0)
#S(frame 0 NIL 0 0 1080 1920)
STUMPWM> (output->head 466 0)
NIL
STUMPWM> (output->head 467 0)
#S(frame 0 NIL 1080 0 2560 1440)
STUMPWM> (output->head 479 0)
NIL
STUMPWM> (make-screen-randr-heads (screen-root cs))
(#S(frame 0 NIL 0 0 1080 1920) #S(frame 2 NIL 1080 0 2560 1440))
STUMPWM> (xlib:rr-get-output-info *display* 446 (get-universal-time))
:SUCCESS
6443810
443
509
286
:CONNECTED
:UNKNOWN
(1245188 1 28966919 29032448)
1
(29097984 29163520 29294592 29360128 29425664 29491200 29556736 29622272
 29687808 29753344 29818880 29884416 29949952 30015488 30081024 30146560
 30212096 30277632 30343168)
NIL
"\0ǐ\0Ǒ\0噄ⵉ"
```
# repl when in single screen mode
```
STUMPWM> (xlib:rr-get-output-info *display* 446 (get-universal-time))
:SUCCESS
6394587
0
510
290
:CONNECTED
:UNKNOWN
(1245188 1 28966919 29032448)
1
(29097984 29163520 29294592 29360128 29425664 29491200 29556736 29622272
 29687808 29753344 29818880 29884416 29949952 30015488 30081024 30146560
 30212096 30277632 30343168)
NIL
"\0ǐ\0Ǒ\0噄ⵉ"

```

**Comment by sdilts** [comment]
Stumpwm also crashes for me when adding and removing  monitors.

**Comment by PuercoPop** [comment]
@sdilts Does the workaround described by @scottjad works for you?

**Comment by sdilts** [comment]
That seems to fix it. After digging a little bit, I found [this stackexchange answser](https://superuser.com/a/1078359) that says that CRTCs are labels for video buffers, and it appears that in my case, each video buffer corresponds to a different output port on my laptop. For me, CRTC 0 corresponds to my laptop's built in display. Is this the case for you, @scottjad? You can get the CRTC numbers for you display with `xrandr --verbose`.

**Comment by sdilts** [comment]
It appears that CRTCs are assigned dynamically, as attaching my display port and VGA connectors results in different numbers depending on which display was attached first. I'd be curious to see what occurs when *all* displays are disconnected. I don't have a setup to test this, however.

**Comment by scottjad** [comment]
@sdilts I think the value of CRTC when calling xrandr --verbose is different than the value referred to in the function `output->head`.

I get CRTC values like 0 and 1 for enabled displays when calling xrandr. In output->head, the crtc values will be like 442 for enabled displays and 0 for disabled displays.

You can evaluate this code in slime to see the values of crtc for your displays:

```
(multiple-value-bind (_1 _2 _3 outputs) (xlib:rr-get-screen-resources (screen-root (current-screen))) (mapcar (lambda (o) (multiple-value-bind (_1 _2 crtc) (xlib:rr-get-output-info *display* o (get-universal-time)) crtc)) outputs))
```
I think you'll find when you go from two to one enabled display, there will be an additional 0 in the list.

To see additional information about the outputs:
```
(multiple-value-bind (_1 _2 _3 outputs) (xlib:rr-get-screen-resources (screen-root (current-screen))) (mapcar (lambda (o) (multiple-value-list (xlib:rr-get-output-info *display* o (get-universal-time)))) outputs))
```

**Comment by zonotope** [comment]
Stump also crashes for me whenever I plug in my external monitor, and it just started happening after a recent update. I had not updated Stump for a long while before then. 

I'm using a display manager, so upon crashes, I'm completely logged out and have to log back in. Often, when I log back in with the external monitor still connected, it crashes again, and again. When I'm finally able to log in without a crash, my monitor configuration is a bit wonky, but I can reconfigure it with lxrandr. The only problem with that is that after I get the configuration I want (where my laptop screen and external monitors mirror each other) and save, stumpwm crashes again and restarts, only this time I'm not logged out of my display manager and am able to capture the stacktrace from the previous error. 

I think my crashes after adjusting the display config with lxrandr are related to those described here, but I have a different stacktrace. I've pasted it below in case it helps track down this issue. I'm happy to open another issue instead if this turns out to be unrelated. 

### Crash Stack Trace
```
The index 1 is too large.
Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10005705B3}>
0: (STUMPWM::BACKTRACE-STRING)
1: ((:METHOD STUMPWM::HANDLE-TOP-LEVEL-CONDITION (SERIOUS-CONDITION)) #<SB-KERNEL:INDEX-TOO-LARGE-ERROR expected-type: (INTEGER 0 0) datum: 1>) [fast-method]
2: (SB-KERNEL::%SIGNAL #<SB-KERNEL:INDEX-TOO-LARGE-ERROR expected-type: (INTEGER 0 0) datum: 1>)
3: (ERROR SB-KERNEL:INDEX-TOO-LARGE-ERROR :DATUM 1 :EXPECTED-TYPE (INTEGER 0 0))
4: (ELT (((#S(frame 1 NIL 0 0 1320 1403) (#S(frame 2 #S(TILE-WINDOW "Display Settings" #x1E00008) 1320 0 1240 961) (#S(frame 3 NIL 1320 961 620 442) #S(frame 4 NIL 1940 961 620 442)))) #S(frame 0 #S(TILE-WINDOW "conky (ursula)" #x1400002) 0 1403 2560 37))) 1)
5: ((:METHOD GROUP-RESIZE-HEAD (STUMPWM::TILE-GROUP T T)) #<STUMPWM::TILE-GROUP {1003055233}> #S(frame 4 NIL 2560 0 2560 1440) #S(frame 4 NIL 0 0 2560 1440)) [fast-method]
6: (STUMPWM::SCALE-HEAD #S<screen #<XLIB:SCREEN :0.0 5120x1440x24 TRUE-COLOR>> #S(frame 4 NIL 2560 0 2560 1440) #S(frame 4 NIL 0 0 2560 1440))
7: (STUMPWM::SCALE-SCREEN #S<screen #<XLIB:SCREEN :0.0 5120x1440x24 TRUE-COLOR>> (#S(frame 0 NIL 0 0 2560 1440) #S(frame 4 NIL 0 0 2560 1440)))
8: (STUMPWM::HEAD-FORCE-REFRESH #S<screen #<XLIB:SCREEN :0.0 5120x1440x24 TRUE-COLOR>> (#S(frame 0 NIL 0 0 2560 1440) #S(frame 4 NIL 0 0 2560 1440)))
9: ((LAMBDA (&REST #:G0 &KEY :STACK-MODE :WINDOW :X :Y :WIDTH :HEIGHT :BORDER-WIDTH :VALUE-MASK &ALLOW-OTHER-KEYS) :IN "/home/zonotope/.cache/yay/stumpwm-git/src/stumpwm/events.lisp") :STACK-MODE NIL :WINDOW #<XLIB:WINDOW :0 1B5> :X 0 :Y 0 :WIDTH 5120 :HEIGHT 1440 :BORDER-WIDTH 0 :VALUE-MASK NIL)
10: (STUMPWM::HANDLE-EVENT :DISPLAY #<XLIB:DISPLAY :0 (The X.Org Foundation R12004000)> :EVENT-KEY :CONFIGURE-NOTIFY :EVENT-CODE 22 :SEND-EVENT-P NIL :SEQUENCE 1998 :EVENT-WINDOW #<XLIB:WINDOW :0 1B5> :WINDOW #<XLIB:WINDOW :0 1B5> :ABOVE-SIBLING NIL :X 0 :Y 0 :WIDTH 5120 :HEIGHT 1440 :BORDER-WIDTH 0 :OVERRIDE-REDIRECT-P NIL)
11: ((FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN XLIB:PROCESS-EVENT))
12: ((FLET "WITHOUT-INTERRUPTS-BODY-29" :IN SB-THREAD::CALL-WITH-RECURSIVE-LOCK))
13: (SB-THREAD::CALL-WITH-RECURSIVE-LOCK #<CLOSURE (FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN XLIB:PROCESS-EVENT) {7FBBA572F06B}> #<SB-THREAD:MUTEX "CLX Event Lock" owner: #<SB-THREAD:THREAD "main thread" RUNNING {10005705B3}>> T NIL)
14: ((FLET SB-IMPL::TIMEOUT-BODY :IN XLIB:PROCESS-EVENT))
15: (XLIB:PROCESS-EVENT #<XLIB:DISPLAY :0 (The X.Org Foundation R12004000)> :HANDLER #<FUNCTION STUMPWM::HANDLE-EVENT> :TIMEOUT 0 :PEEK-P NIL :DISCARD-P NIL :FORCE-OUTPUT-P T)
16: ((FLET "WITHOUT-INTERRUPTS-BODY-29" :IN SB-THREAD::CALL-WITH-RECURSIVE-LOCK))
17: (SB-THREAD::CALL-WITH-RECURSIVE-LOCK #<CLOSURE (FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN STUMPWM::DISPATCH-ALL) {7FBBA572F4BB}> #<SB-THREAD:MUTEX "CLX Event Lock" owner: #<SB-THREAD:THREAD "main thread" RUNNING {10005705B3}>> T NIL)
18: ((FLET STUMPWM::DISPATCH-ALL :IN "/home/zonotope/.cache/yay/stumpwm-git/src/stumpwm/stumpwm.lisp") #<XLIB:DISPLAY :0 (The X.Org Foundation R12004000)>)
19: ((:METHOD IO-LOOP (STUMPWM::SBCL-IO-LOOP)) #<STUMPWM::SBCL-IO-LOOP {1003578493}> :DESCRIPTION "StumpWM") [fast-method]
20: (STUMPWM::STUMPWM-INTERNAL-LOOP)
21: (STUMPWM::STUMPWM-INTERNAL ":0")
22: (STUMPWM ":0")
23: ((LAMBDA NIL :IN "/home/zonotope/.cache/yay/stumpwm-git/src/stumpwm/make-image.lisp"))
24: ((FLET SB-UNIX::BODY :IN SB-EXT:SAVE-LISP-AND-DIE))
25: ((FLET "WITHOUT-INTERRUPTS-BODY-14" :IN SB-EXT:SAVE-LISP-AND-DIE))
26: ((LABELS SB-IMPL::RESTART-LISP :IN SB-EXT:SAVE-LISP-AND-DIE))
```

**Comment by scottjad** [comment]
Fixed by #611.
